streaming-binlog/src/main/java/org/openmrs/analytics/DebeziumListener.java

```diff
@@ -66,17 +73,27 @@ FhirConverter createFhirConverter(CamelContext camelContext) throws IOException,
 		return new FhirConverter(openmrsUtil, fhirStoreUtil, parquetUtil, params.fhirDebeziumEventConfigPath);
 	}
 	
-	private String getDebeziumConfig() {
-		return "debezium-mysql:" + params.databaseHostName + "?" + "databaseHostname=" + params.databaseHostName
-		        + "&databaseServerId=" + params.databaseServerId + "&databasePort=" + params.databasePort.intValue()
-		        + "&databaseUser=" + params.databaseUser + "&databasePassword=" + params.databasePassword
-				//+ "&name={{database.dbname}}"
-		        + "&databaseServerName=" + params.databaseName + "&databaseWhitelist=" + params.databaseSchema
-		        + "&offsetStorage=org.apache.kafka.connect.storage.FileOffsetBackingStore" + "&offsetStorageFileName="
-		        + params.databaseOffsetStorage + "&databaseHistoryFileFilename=" + params.databaseHistory + "&snapshotMode="
-		        + params.snapshotMode
-		//+ "&tableWhitelist={{database.schema}}.encounter,{{database.schema}}.obs"
-		;
+	private String getDebeziumConfig() throws IOException {
+		GeneralConfiguration generalConfiguration = loadDebeziumConfig();
+		DebeziumConfiguration debeziumConfigs = generalConfiguration.getDebeziumConfigurations();
+		return "debezium-mysql:" + debeziumConfigs.getDatabaseHostName() + "?" + "databaseHostname="
+		        + debeziumConfigs.getDatabaseHostName() + "&databaseServerId=" + debeziumConfigs.getDatabaseServerId()
+		        + "&databasePort=" + debeziumConfigs.getDatabasePort() + "&databaseUser=" + debeziumConfigs.getDatabaseUser()
+		        + "&databasePassword=" + debeziumConfigs.getDatabasePassword() + "&databaseServerName="
+		        + debeziumConfigs.getDatabaseName() + "&databaseWhitelist=" + debeziumConfigs.getDatabaseSchema()
+		        + "&offsetStorage=" + debeziumConfigs.getOffsetStorage() + "&offsetStorageFileName="
+		        + debeziumConfigs.getDatabaseOffsetStorage() + "&databaseHistoryFileFilename="
+		        + debeziumConfigs.getDatabaseHistory() + "&snapshotMode=" + debeziumConfigs.getSnapshotMode();
+	}
+	
+	@VisibleForTesting
+	GeneralConfiguration loadDebeziumConfig() throws IOException {
```

Comment:
Does this actually load a full `GeneralConfiguration`? It seems that this method only fills the `debeziumConfiguration` part right? (I understand it is enough for the use-case here but it is a little bit confusing.)

I have made some suggestions below to merge the two JSON files. Once you do that, please also move this function to be a static factory method inside `GeneralConfiguration` and use it everywhere that we need a GeneralConfiguration instance.